nifi-nar-bundles/nifi-dbcp-services-bundle/nifi-postgres-dbcp-service/src/main/java/org/apache/nifi/service/PostgreSQLConnectionPool.java

```diff
@@ -0,0 +1,150 @@
+            .displayName("Connection URL Format")
+            .description("The format of the connection URL.")
+            .allowableValues(JdbcUrlFormat.class)
+            .required(true)
+            .defaultValue(FULL_URL.getValue())
+            .build();
+
+    public static final PropertyDescriptor POSTGRES_DATABASE_URL = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(DATABASE_URL)
+            .dependsOn(CONNECTION_URL_FORMAT, FULL_URL)
+            .build();
+
+    public static final PropertyDescriptor DATABASE_NAME = new PropertyDescriptor.Builder()
+            .name("database-name")
+            .displayName("Database Name")
+            .description("The name of the database to connect.")
+            .required(false)
+            .dependsOn(CONNECTION_URL_FORMAT, PARAMETERS)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    public static final PropertyDescriptor DATABASE_HOSTNAME = new PropertyDescriptor.Builder()
+            .name("database-hostname")
+            .displayName("Database Hostname")
+            .description("The hostname of the database to connect.")
+            .required(false)
+            .dependsOn(CONNECTION_URL_FORMAT, PARAMETERS)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    public static final PropertyDescriptor DATABASE_PORT = new PropertyDescriptor.Builder()
+            .name("database-port")
+            .displayName("Database Port")
+            .description("The port of the database to connect.")
+            .required(false)
+            .dependsOn(CONNECTION_URL_FORMAT, PARAMETERS)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(
+            CONNECTION_URL_FORMAT,
+            POSTGRES_DATABASE_URL,
+            DATABASE_NAME,
+            DATABASE_HOSTNAME,
+            DATABASE_PORT,
+            DB_USER,
+            DB_PASSWORD,
+            KERBEROS_USER_SERVICE,
+            MAX_WAIT_TIME,
+            MAX_TOTAL_CONNECTIONS,
+            VALIDATION_QUERY,
+            MIN_IDLE,
+            MAX_IDLE,
+            MAX_CONN_LIFETIME,
+            EVICTION_RUN_PERIOD,
+            MIN_EVICTABLE_IDLE_TIME,
+            SOFT_MIN_EVICTABLE_IDLE_TIME
+    ));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    protected Driver getDriver(final String driverName, final String url) {
+        try {
+            return DriverManager.getDriver(url);
+        } catch (Exception e) {
+            throw new ProcessException("PostgreSQL driver unavailable or incompatible connection URL", e);

```

Comment:
This should only ever happen if they entered an URL in invalid format (certainly not by entering parameters as we construct the URL for the user), recommend adding a unit test for this.